### ISSUE TITLE
Inaccurate and unclear pronunciation of Chinese numeric amounts

### ISSUE THREAD

**Issue by tyftql** [OpenBMB/VoxCPM#164]
Hi VoxCPM Team,
Thank you very much for open-sourcing VoxCPM and providing the detailed technical report. The Causal Audio VAE architecture is impressive, and I'm excited to test its TTS capabilities for Chinese scenarios.
However, during my testing, I found a critical issue: the model's pronunciation of Chinese numeric amounts (especially integers with decimals) is inaccurate and unclear (muffled/hazy), making it difficult to identify the exact numbers from the generated audio.
I have conducted multiple tests with different numeric amounts, and the problem persists even after adjusting key parameters (such as cfg_value, inference_timesteps, denoise). Below are the detailed reproduction steps, test cases, and complete test script for your reference.
case text：
1228元
1356.61元
356.61元
这个月房租是1228元

import os
import torch
import soundfile as sf
import numpy as np

os.environ["VOXCPM_SKIP_AUTO_DOWNLOAD"] = "1"
os.environ["HF_HUB_DISABLE_DOWNLOAD"] = "1"
os.environ["HF_HUB_OFFLINE"] = "1"
os.environ["MODELSCOPE_HUB_OFFLINE"] = "1"
os.environ["MODELSCOPE_CACHE"] = "/data04/j-tangyufei-jk/VoxCPM/"

os.environ["VOXCPM_ZIPENHANCER_MODEL_PATH"] = "/data04/j-tangyufei-jk/VoxCPM/enhancer_zip"
os.environ["VOXCPM_DENOISER_MODEL_PATH"] = "/data04/j-tangyufei-jk/VoxCPM/enhancer_sensevoice"

os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
os.environ["TORCH_CUDNN_V8_API_ENABLED"] = "1"

from voxcpm import VoxCPM  # 官方上层封装类
MODEL_PATH = "/data04/j-tangyufei-jk/VoxCPM/VoxCPM1.5"
PROMPT_WAV_PATH = "XXXXXX"
PROMPT_TEXT = "XXXXXX"
TARGET_TEXT = "这个月房租是1228元
OUTPUT_PATH_STREAMING = "/data04/j-tangyufei-jk/VoxCPM/streaming_result.wav"  # 流式音频保存路径

def save_streaming_audio(wav_chunks: list, sample_rate: int, output_path: str):
    """保存流式生成的音频块为完整 WAV 文件"""
    # 拼接所有流式音频块
    wav_np = np.concatenate(wav_chunks)
    # 创建输出目录（若不存在）
    output_dir = os.path.dirname(output_path)
    if not os.path.exists(output_dir):
        os.makedirs(output_dir, exist_ok=True)
    # 保存为 WAV 文件（失败则降级为 FLAC）
    try:
        sf.write(output_path, wav_np, sample_rate, format='WAV', subtype='PCM_16')
        print(f"\n✅ 流式音频保存成功：{output_path} | 音频长度：{len(wav_np)/sample_rate:.2f} 秒")
    except Exception as e:
        fallback_path = output_path.replace('.wav', '.flac')
        sf.write(fallback_path, wav_np, sample_rate, format='FLAC')
        print(f"\n⚠️ WAV 格式保存失败，已降级为 FLAC：{fallback_path}")

if __name__ == "__main__":
    gpu_id = 0
    os.environ["CUDA_VISIBLE_DEVICES"] = str(gpu_id)
    torch.cuda.set_device(gpu_id)
    torch.cuda.empty_cache()
    print(f"📌 已绑定 GPU{gpu_id} | 型号：{torch.cuda.get_device_name(gpu_id)}")

    print("\n📥 加载本地 VoxCPM1.5 模型...")
    model = VoxCPM.from_pretrained(MODEL_PATH)
    torch.cuda.empty_cache()
    print("✅ 模型加载完成（使用内部默认精度，避免输入与参数类型不匹配）")

    print("\n🚀 开始流式生成音频...")
    streaming_chunks = []
    with torch.no_grad():  # 禁用梯度计算，减少显存占用
        for chunk in model.generate_streaming(
            text=TARGET_TEXT,
            prompt_wav_path=PROMPT_WAV_PATH,
            prompt_text=PROMPT_TEXT,
            min_len=2,
            max_len=8192,
            inference_timesteps=10,
            cfg_value=2,
            denoise=False,  
            normalize=False
        ):
            streaming_chunks.append(chunk)  # 收集每个流式音频块
            current_total_len = len(np.concatenate(streaming_chunks)) / model.tts_model.sample_rate
            print(f"🔹 已接收音频块，当前累计长度：{current_total_len:.2f} 秒")

    save_streaming_audio(streaming_chunks, model.tts_model.sample_rate, OUTPUT_PATH_STREAMING)

    torch.cuda.empty_cache()
    print("\n🎉 所有操作完成！请查看保存的音频文件测试效果。")

**Comment by liuxin99** [comment]
Hi, thank you for your testing and for bringing this issue to our attention. At the current stage, we haven't applied ITN (Inverse Text Normalization) to our training data, so the model’s performance on numerical pronunciation is not yet optimal. Therefore, we recommend performing text normalization on the input before feeding it into the model to resolve this issue.
